packages/button/src/Button.story.tsx

```diff
@@ -19,36 +22,18 @@ const meta: StoryMetaType<typeof Button> = {
     default: 'Playground',
   },
   argTypes: {
-    disabled: {
-      control: { type: 'boolean' },
-    },
-    leftGlyph: {
-      options: Object.keys(glyphs),
-      control: { type: 'select' },
-    },
-    rightGlyph: {
-      options: Object.keys(glyphs),
-      control: { type: 'select' },
-    },
-    type: {
-      control: 'select',
-      options: ['button', 'submit'],
-      defaultValue: 'button',
-    },
-    size: {
-      control: 'select',
-      options: Object.values(Size),
-      defaultValue: Size.Default,
-    },
-    href: {
-      control: 'text',
-    },
+    disabled: StorybookArgTypes.Boolean(),
+    leftGlyph: StorybookArgTypes.Select(Object.keys(glyphs)),
+    rightGlyph: StorybookArgTypes.Select(Object.keys(glyphs)),
+    type: StorybookArgTypes.Select(['button', 'submit']),
+    size: StorybookArgTypes.Select(Object.values(Size)),
+    href: StorybookArgTypes.Text(),
```

Comment:
I like this idea a lot - this obviously looks a lot cleaner. My only concern is that because it's another layer of logic to maintain that might be very hard to track down errors in, I'd want these object-transforming utilities to be pretty heavily tested.

Reply:
I'm with Sean on this